libraries/plugins/blockchain_history/blockchain_history_api.cpp

```diff
@@ -95,6 +95,57 @@ class blockchain_history_api_impl
         return result;
     }
 
+    template <typename IndexType>
+    result_type get_ops_history_by_time(const fc::time_point_sec& from,
+                                        const fc::time_point_sec& to,
+                                        uint32_t from_op,
+                                        uint32_t limit) const
+    {
+        FC_ASSERT(from <= to, "'From' is greater than 'to'");
+        FC_ASSERT((to - from).to_seconds() <= get_api_config(API_BLOCKCHAIN_HISTORY).max_timestamp_range_in_s,
+                  "Timestamp range can't be more then ${t} seconds",
+                  ("t", get_api_config(API_BLOCKCHAIN_HISTORY).max_timestamp_range_in_s));
+        FC_ASSERT(limit > 0, "Limit must be greater than zero");
+        FC_ASSERT(limit <= get_api_config(API_BLOCKCHAIN_HISTORY).max_blockchain_history_depth,
+                  "Limit of ${l} is greater than maxmimum allowed ${2}",
+                  ("l", limit)("2", get_api_config(API_BLOCKCHAIN_HISTORY).max_blockchain_history_depth));
+        FC_ASSERT(from_op >= limit, "From must be greater than limit");
+
+        result_type result;
+
+        const auto& idx = _db->get_index<IndexType>().indices().get<by_timestamp>();
+        if (idx.empty())
+            return result;
+
+        auto range = idx.range(::boost::lambda::_1 >= std::make_tuple(from, 0),
+                               ::boost::lambda::_1 <= std::make_tuple(to, ALL_IDS));
+
+        using operation_object_ref_type = std::reference_wrapper<const operation_object>;
+        using op_ids_type = std::vector<operation_object_ref_type>;
+
+        op_ids_type ids;
+
+        for (auto it = range.first; it != range.second; ++it)
```

Comment:
This cycle is not needed. You can use range at once without temporary array of ids